drivers/video/fbdev/bcm2708_fb.c

```diff
@@ -883,87 +1034,149 @@ static int bcm2708_fb_probe(struct platform_device *dev)
 	if (!fw)
 		return -EPROBE_DEFER;
 
-	fb = kzalloc(sizeof(*fb), GFP_KERNEL);
-	if (!fb) {
-		ret = -ENOMEM;
-		goto free_region;
+	ret = rpi_firmware_property(fw,
+				    RPI_FIRMWARE_FRAMEBUFFER_GET_NUM_DISPLAYS,
+				    &num_displays, sizeof(u32));
+
+	/* If we fail to get the number of displays, or it returns 0, then
+	 * assume old firmware that doesn't have the mailbox call, so just
+	 * just set one display
+	 */
+	if (ret || num_displays == 0) {
+		num_displays = 1;
+		old = 1;
+		dev_err(&dev->dev,
+			"Unable to determine number of FB's. Assuming 1\n");
+		print_debug("Unable to determine number of FB's. Assuming 1\n");
+		ret = 0;
 	}
 
-	fb->fw = fw;
-	bcm2708_fb_debugfs_init(fb);
+	dev_info(&dev->dev, "FB found %d display(s)\n", num_displays);
 
-	fb->cb_base = dma_alloc_writecombine(&dev->dev, SZ_64K,
-					     &fb->cb_handle, GFP_KERNEL);
-	if (!fb->cb_base) {
-		dev_err(&dev->dev, "cannot allocate DMA CBs\n");
-		ret = -ENOMEM;
-		goto free_fb;
-	}
+	/* Set up the DMA information. Note we have just one set of DMA for
+	 * all the FB's, and simply ensure that the DMA only gets called
+	 * by one at a time
+	 */
+	if (!fbdev->cb_base) {
+		fbdev->cb_base = dma_alloc_writecombine(&dev->dev, SZ_64K,
+							&fbdev->cb_handle,
+							GFP_KERNEL);
+		if (!fbdev->cb_base) {
+			dev_err(&dev->dev, "cannot allocate DMA CBs\n");
+			ret = -ENOMEM;
+			goto free_fb;
+		}
 
-	pr_info("BCM2708FB: allocated DMA memory %pad\n", &fb->cb_handle);
+		dev_info(&dev->dev,
+			 "Allocated DMA memory %08x\n",
+			 fbdev->cb_handle);
```

Comment:
You've also reverted a cleanup.
The log line should end ```%pad\n", &fbdev->cb_handle);``` as it is a dma_addr_t which is size dependent and passed by reference.

Reply:
Bugger, thought I had got all those sorted. Soz. It was a nasty merge.